e2e/crd_validation_test.go

```diff
@@ -96,6 +97,241 @@ func cleanupKindCluster(t *testing.T, clusterName string) func() {
 	}
 }
 
+func TestCRDDefaulting(t *testing.T) {
+	t.Parallel()
+
+	c := createKindCluster(t)
+
+	type cpmTest struct {
+		obj  *monitoringv1.ClusterPodMonitoring
+		want *monitoringv1.ClusterPodMonitoring
+	}
+	runCPM := func(t *testing.T, tests map[string]cpmTest) {
```

Comment:
question(non-blocking): this seems to deviate a bit from vanilla [table-driven testing](https://dave.cheney.net/2013/06/09/writing-table-driven-tests-in-go), which is much more common in the codebase.

Is there a reason for not going with a pattern like [this](https://github.com/GoogleCloudPlatform/prometheus-engine/blob/bc30e6a2200ee3c531ba8c4a208256483c2e1a06/cmd/rule-evaluator/main_test.go#L78)? I find it a bit more conventional and easier to read.

Reply:
Ok - I think it's ultimately a style thing. My preference is for more bite-sized test methods (e.g. break out CPM tests into a separate function altogether). I generally find that more reviewable/readable. But it's subjective, so I won't block.

Thanks for the links!